File: code/jvm/src/main/kotlin/pt/isel/talkRooms/domain/Channel.kt
package pt.isel.talkRooms.domain

data class Channel(
    val id: Int,
    val name: String,
    val creator: User,
    val visibility: Visibility,
    val admins: User, //TODO one or more??
    val numberOfUsers: Int,
    val messages: List<Message>,
    val users: Map<User, Role>
)

File: code/js/src/domain/Channel.ts
import { User } from "./User";
import { Visibility } from "./Visibility";

/*
export type Channel = {
    id: number,
    name: string,
    creator: User,
    visibility: Visibility,
}*/

export class Channel {
    id: number;
    name: string;
    creator: User;
    visibility: Visibility;

    constructor(id: number, name: string, creator: User, visibility: Visibility) {
        this.id = id;
        this.name = name;
        this.creator = creator;
        this.visibility = visibility;
    }

    public equals(other: Channel): boolean {
        return this.id === other.id && this.name === other.name && this.creator.equals(other.creator) && this.visibility === other.visibility;
    }
}